### ISSUE TITLE
isSideLoaded issue tag on Android not working as expected

### ISSUE THREAD

**Issue by Nohus** [getsentry/sentry-java#1906]
The `isSideLoaded` tag for Android displays `false` even for sideloaded apps (apps installed directly from the APK):

![image](https://user-images.githubusercontent.com/11064768/153629916-df6aa7ae-9e6d-4461-af43-6819246a1871.png)

Here the `installerStore` tag shows the app was installed by `com.android.packageinstaller`, which is the system app used when installing an APK manually, and yet `isSideLoaded` shows `false`.

@marandaneto identified the Android documentation is misleading here: https://developer.android.com/reference/android/content/pm/PackageManager#getInstallerPackageName(java.lang.String)
It says "Retrieve the package name of the application that installed a package. This identifies which market the package came from."
The first sentence is true, the second is not, since not all installer apps are "markets".

Ultimately I don't think the `isSideLoaded` tag is useful even with this fixed, since it's not really defined what "sideloaded" means. Even when the app is actually installed from a market, there are shady market apps that rehost apps copied from Google Play. And apps installed from them are definitely "installed from a store/market", but the information useful for the app developer is that it was installed from an unofficial source, even though it was not technically "sideloaded" (or was it? where is the line?).

What I would propose as the actual fix:
- Remove the `isSideLoaded` tag: it doesn't provide any information not already visible in the `installerStore` tag
- Rename `installerStore` to `installer`: it's not always a store
- Use the red highlight (as visible in `isSideLoaded: false` on the screenshot) on the `installer` tag, when it doesn't contain one of the reputable stores (Google Play, Samsung Galaxy Store, Amazon Appstore)
- Show the `installer` as `none` when none

**Comment by Nohus** [comment]
While I agree that my proposed solution is an enhancement going further than the original issue, this issue is still a bug: `isSideLoaded` shows `false` for sideloaded apps.

**Comment by bruno-garcia** [comment]
Thanks for raising this. We'll dig into the incorrect value for isSideLoaded further and update this issue

**Comment by bradjones1** [comment]
This issue also affects the React Native/Expo Sentry packages.

**Comment by markushi** [comment]
@Nohus is this still an issue for you? In case it is please provide a repro and we're happy to look into this again.

**Comment by Nohus** [comment]
Yes, this bug is still present and the reproduction steps are still as in the original comment: download an APK, install it, and when an issue is reported to Sentry it shows as not sideloaded, even though it is.

**Comment by marandaneto** [comment]
Libs do the same as us:
https://github.com/javiersantos/PiracyChecker/blob/4ebafd4890593a09a2b978ed38e6dc7dc6ce79e1/library/src/main/java/com/github/javiersantos/piracychecker/utils/LibraryUtils.kt#L93-L104

The difference is that they also account for the known stores, eg https://github.com/javiersantos/PiracyChecker/blob/4ebafd4890593a09a2b978ed38e6dc7dc6ce79e1/library/src/main/java/com/github/javiersantos/piracychecker/enums/InstallerID.kt#L6-L9

Testing locally, `getInstallerPackageName` always returns null if it's installed via `adb`.
`getInstallSourceInfo` requires `INSTALL_PACKAGES` permission so not really useful (I guess?).

I guess what we can do here is to special case `com.android.packageinstaller` and consider it as `isSideLoaded=true` in this case instead of only `null`, maybe we'll find out more package names, we can fix 1 by 1.

The best idea would be to use something like https://github.com/javiersantos/PiracyChecker#verify-google-play-licensing-lvl but this does not work everywhere (eg China) anyway.

**Comment by marandaneto** [comment]
I just did `adb push` from the computer to the emulator, installed the app via the file explorer, and still getting `installerStore=true` correctly.

@Nohus how do you reproduce this problem?

> installing an APK manually

Can you tell me how you install the APK?

**Comment by Nohus** [comment]
The linked approach from PiracyChecker looks to be what I suggested above;
having a list of valid stores and consider everything else sideloaded. That
makes sense to me.

I installed the app from Chrome's Downloads list, which started the package
installer. Alternatively you can install any unofficial app store like
ApkPure and install from there. It has rehosted copies of apps from Google
Play.

**Comment by Nohus** [comment]
> I guess what we can do here is to special case
com.android.packageinstaller and consider it as isSideLoaded=true

That wouldn't account for all the other apps that can install stolen copies
of apps like ApkPure or ApkMirror, or backup apps restoring apps from a
backup (reinstalling a backed up APK).

**Comment by marandaneto** [comment]
Gotcha, I will leave the Android team to handle this then, thanks for the feedback @Nohus

**Comment by markushi** [comment]
Let's go for solution 2, as it should be easier to maintain and wouldn't cause a breaking change (e.g. considering self-hosted).

**Comment by kahest** [comment]
related: https://github.com/getsentry/sentry-java/issues/3026
